test/kube2e/helper/curl.go

```
@@ -65,13 +64,17 @@ func (t *testContainer) CurlEventuallyShouldOutput(opts CurlOpts, substr string,
 	tick := time.Tick(currentTimeout / 8)
 
 	gomega.EventuallyWithOffset(ginkgoOffset+1, func() string {
+		if !t.CanCurl() {
+			return ""
```

We might want to return an error from this func, so we can return an error in this case and make clear what's happening

From [gomega docs on `Eventually()`](https://pkg.go.dev/github.com/onsi/gomega#Eventually):
>If multiple values are returned by the function, Eventually will pass the first value to the matcher and require that all others are zero-valued. This allows you to pass Eventually a function that returns a value and an error - a common pattern in Go.

---

Went ahead and asserted that we can curl with the outer gomega, therefore failing if false.